examples/deprecation/deprecation.bal

```diff
@@ -0,0 +1,59 @@
+// The constant `CITY` is marked as deprecated using the `@deprecated` annotation and the markdown documentation `# # Deprecated`.
+# Define a constant to represent the city.
+# # Deprecated
+# This constant is deprecated as this information is not needed.
+@deprecated
+const string CITY = "COLOMBO";
+const string COUNTRY = "Sri Lanka";
+
+// The type `Address` is marked as deprecated using the `@deprecation` annotation.
+@deprecated
+public type Address CITY|COUNTRY;
+
+// The object `Person` is marked as deprecated using `@deprecated` annotation and the markdown documentation `# # Deprecated`.
+# The `Person` is a user-defined object.
+#
+# + firstName - The first name of the person.
+# + lastName - The last name of the person.
+# # Deprecated
+# This object is deprecated as information representation is changed.
+# Users of this object should instead use the object `StudentData`.
+@deprecated
+public type Person object {
+    public string firstName = "Andrew";
+    public string lastName = "John";
+    // Usage of deprecated type 'Address' and deprecated constant `CITY`.
+    Address addr = CITY;
+
+    // The object function `updateFirstName` is deprecated using `@deprecation` annotation.
```

Comment:
**Object function** is the keyword used in spec to define the deprecation annotation. Therefore shall we keep it as **function**

Reply:
That's just for the annotation declaration, probably so that we don't have to introduce another keyword `method`.

I think we should change this to method.

Reply:
Done